provisioning_templates/cloud_init/cloud_init_default.erb

```diff
@@ -27,6 +27,7 @@ users: {}
 <% if puppet_enabled %>
 runcmd:
 - |
+<%= snippet 'redhat_register' %>
```

Comment:
Late review, but note that this is behind `<% if puppet_enabled %>`. I doubt you need Puppet to do RH register. Also note that this template is used on non-RH systems as well but I think the host parameter checks inside that take care of it.

Reply:
Damn! Good find, @san7ket can you do another PR and fix this? Just move it out of the puppet_enabled block.

> I think the host parameter checks inside that take care of it.

Yep, I have checked that. Can confirm.

Reply:
ahh yes, will raise a fix

Reply:
Raised fix in #611